test/Api.Tests/TestStartup.cs

```diff
@@ -14,7 +12,9 @@ public void ConfigureTestServices(IServiceCollection services)
             base.ConfigureServices(services);
 
             // Register here the dependency injection services
-           // services.AddDbContext<ApplicationDbContext>(opts => opts.UseInMemoryDatabase());
```

Comment:
It sounds like you did't understand the differences between the mocking you are using, and the mocking form the tutorial. You're mocking the dbcontext on MVC level. At the tutorial they are using an InMemoryDatabase to test also the EF Core layer (which i already told you would be better for our purpose). 
And you do a mixture! But if you look to the tutorial, they are inherit form the StartUp, right. But with EF Core and the have some Exceptions for testing in the StartUp especially for the InMemoryDatabase.